src/libs/Navigation/AppNavigator/ReportScreenIDSetter.ts

```diff
@@ -57,11 +37,18 @@ const getLastAccessedReportID = (
     return lastReport?.reportID;
 };
 
-// This wrapper is reponsible for opening the last accessed report if there is no reportID specified in the route params
-function ReportScreenIDSetter({route, reports, policies, navigation, isFirstTimeNewExpensifyUser = false, reportMetadata, accountID, personalDetails}: ReportScreenIDSetterProps) {
+// This wrapper is responsible for opening the last accessed report if there is no reportID specified in the route params
+function ReportScreenIDSetter({route, navigation}: ReportScreenIDSetterProps) {
     const {canUseDefaultRooms} = usePermissions();
     const {activeWorkspaceID} = useActiveWorkspace();
 
+    const [reports] = useOnyx(ONYXKEYS.COLLECTION.REPORT, {allowStaleData: true});
+    const [policies] = useOnyx(ONYXKEYS.COLLECTION.POLICY, {allowStaleData: true});
+    const [personalDetails] = useOnyx(ONYXKEYS.PERSONAL_DETAILS_LIST);
+    const [isFirstTimeNewExpensifyUser] = useOnyx(ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER, {initialValue: false});
```

Comment:
NAB but you `initialValue` is kind of outdated for `useOnyx`. IMO this is a clearer way to achieve the same thing, and the resultant type is `boolean` instead of `boolean | undefined`.

```suggestion
    const [isFirstTimeNewExpensifyUser = false] = useOnyx(ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER);
```

Reply:
hey thanks, I will keep this in mind moving forward.
This was an experiment in rewriting, so I wanted to be very 1-to-1 in regards to `withOnyx` -> `useOnyx`